App.js

```diff
@@ -68,69 +78,64 @@ export default class App extends React.Component {
       }
     ],
     newPin: {
-      user_id: 'test_userID',
-      title_input: '',
-      details_input: '',
-      type_input: '',
+      title_input: "",
+      details_input: "",
+      type_input: "",
       coordinate: {
         latitude: null,
         longitude: null
-      },
-
+      }
     }
-  }
+  };
 
-  componentWillMount() {
+  async componentDidMount() {
     if (Platform.OS === "android" && !Constants.isDevice) {
       this.setState({
         errorMessage:
-          "Oops, this will not work on Sketch in an Android emulator. Try it on your device!"
+          "Geolocation will not work on Sketch in Android emulator. Try it on your device!"
       });
     } else {
-      this._getLocationAsync();
+      await this._setupUser();
+      await this._getLocationAsync();
     }
   }
 
+  _setupUser() {
+    if (this.state.user_id) {
```

Comment:
Does this actually do anything? `this.state.user_id` starts as `null`, right, so this should not be executed?

Reply:
It's not doing anything and the ID is being passed in as `null`

Reply:
this makes sure that you don't make multiple create user calls to firebase.
there is a fix for id being passed in as null in #37 .